src/main/java/seedu/address/logic/commands/AddToFolderCommand.java

```diff
@@ -48,6 +49,12 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToAdd = lastShownList.get(index.getZeroBased());
 
+        if (!model.hasFolderName(folderName)) {
+            throw new CommandException(MESSAGE_NONEXISTENT_FOLDER);
+        } else if (model.folderContainsPerson(personToAdd, folderName)) {
+            throw new CommandException(MESSAGE_DUPLICATE_CONTACT);
+        }
```

Comment:
Since both scenarios will not occur at the same time, would it be better to convert `else if` to `if` branch instead? I generally find it easier to read. You can decide
i.e.:
```
if (!model.hasFolderName(foldername)) {
    ...
} 
if (model.folderContainsPerson(personToAdd, folderName)) {
   ...
}
```